sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -425,6 +427,13 @@ static void appendCompileOptionsFromImage(std::string &CompileOpts,
     // is fixed
     CompileOpts += isEsimdImage ? "-doubleGRF" : "-ze-opt-large-register-file";
   }
+  if (Img.getDeviceGlobals().size() != 0) {
+    // If the image has device globals we need to add the
+    // -ze-take-global-address option to tell IGC to record addresses of these.
+    if (!CompileOpts.empty())
+      CompileOpts += " ";
+    CompileOpts += "-ze-take-global-address";
```

Comment:
If we decide not to move it, then we should add more checks for the underlying runtime, because I suppose that not all OpenCL runtimes support this argument

Reply:
The plugins do not know which images have device_global values (except CUDA due to the additional program metadata). We could always enable it, but if we can do it here we can at least avoid any potential overhead incurred by the option when it is not needed.
Do you know if there is a good way to identify if IGC is the device compiler to be used?

Reply:
I have updated the heuristic here to check that it's either an OpenCL or L0 backend in use and that all devices are GPUs.